src/metabase/lib/schema/util.cljc

```diff
@@ -129,21 +129,70 @@
        (map? x) (dissoc :lib/uuid)))
    x))
 
-(defn indexed-order-bys-for-stage
-  "Convert all order-bys in a stage to refer to aggregations by index instead of uuid"
-  [{:keys [aggregation order-by] :as stage}]
-  (if (and aggregation order-by)
-    (let [agg-lookups (->> aggregation
-                           (map-indexed (fn [i [_type {agg-uuid :lib/uuid}]]
-                                          [agg-uuid i]))
-                           (into {}))]
-      (update stage :order-by (fn [order-bys]
-                                (mapv (fn [[_dir _opts [order-type agg-opts agg-uuid] :as order-by]]
-                                        (if (= order-type :aggregation)
-                                          (assoc order-by 2 [order-type agg-opts (agg-lookups agg-uuid)])
-                                          order-by))
-                                      order-bys))))
-    stage))
+(defn sorted-maps
+  "Recursively convert all maps in `x` to sorted maps using the same comparator
+   as [[metabase.lib.schema.common/unfussy-sorted-map]]. This ensures consistent
+   JSON serialization order regardless of how maps were constructed."
+  [x sorted-map-fn]
+  (perf/postwalk
+   (fn [x]
+     (if (and (map? x) (not (sorted? x)))
+       (into (sorted-map-fn) x)
+       x))
+   x))
+
+(defn- aggregation-ref?
+  "Check if `x` is an aggregation ref clause `[:aggregation opts uuid-string]`."
+  [x]
+  (and (vector? x)
+       (= :aggregation (first x))
+       (>= (count x) 3)
```

Comment:
are they supposed to be exactly three element tuples or can then actually be longer?